### PR TITLE
Describe processing algorithms

### PR DESCRIPTION
**Pull request by bstroebl**
Description of

1. Dissolve
2. Eliminate sliver polygons
3. Split lines with lines

These were chosen because I either made (2., 3.) or improved them (1.).

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vector_overlay_tools.rst [hunk 1]
```diff
@@ -168,6 +168,40 @@ Console usage
...
+``Input layer`` [vector: line]
+  Line layer containing the lines to be split.
+
+``Split layer`` [vector: line]
+  Line layer whose lines are used to define the breaking points.
+
+Outputs
+.......
+
+``Output layer`` [vector]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_overlay_tools.rst]: Actually, this option is named `splitted` (?) in the GUI.

**Comment by DelazJ** [comment]
Add a minor comment otherwise LGTM. Thanks

**Comment by DelazJ** [comment]
@bstroebl I remark that some of Eliminate sliver polygons tool does not have help associated in QGIS Dektop. Would that be possible to add it?

**Comment by bstroebl** [comment]
@DelazJ Thanks for the hint; I already realised that a description for Eliminate was missing, it has been fixed in https://github.com/qgis/QGIS/commit/4a523511275b48b02592ff4cc64fbf726e4f6547

File: source/docs/user_manual/processing_algs/qgis/vector_overlay_tools.rst (hunk 1 above)
Comment (bstroebl) [inline source/docs/user_manual/processing_algs/qgis/vector_overlay_tools.rst]: Fixed in latest commit, copy-paste error :-)

**Comment by DelazJ** [comment]
ok, nice! Maybe it wasn't backported to master_2 reason why the help is still empty in 2.17 ( qgis/QGIS@f7a1dcd)

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 2]
```diff
@@ -374,24 +383,31 @@ Eliminate sliver polygons
 Description
 ...........
 
-<put algorithm description here>
+This algorithm combines selected polygons of the input layer with certain adjacent polygons
+by erasing their common boundary. Eliminate can either use an
+exisiting selection or a logical query based on one of the layer's fields to make the selection itself.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: existing

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 3]
```diff
@@ -374,24 +383,31 @@ Eliminate sliver polygons
...
+The adjacent polygon can be either the one with the largest or smallest area or the one sharing the
+largest common boundary with the polygon to be eliminated.
+Eliminate is normally used to get rid of sliver polygons, i.e. tiny
+polygons that are a result of polygon instersection processes where boundaries of the inputs
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: intersection

**Comment by DelazJ** [comment]
Sorry, found new typos (while reading the PR in QGIS)

**Comment by bstroebl** [comment]
Are you done reading so I fix them all with one commit :-)
Thanks for pointing it out. Although reading the result several times I did not see them, showing that you only see what you expect to see...

**Comment by DelazJ** [comment]
yes I (re)checked the rest of the text and I think it's ok now.

**Comment by DelazJ** [comment]
Thanks @bstroebl 
Just a last question: have you also made a PR to update `Dissolve` tool help description (extension to line layers)?

**Comment by bstroebl** [comment]
Thanks for merging @DelazJ 
you're right, the help description of `dissolve` should be updated to. Best would be if it has the same wording as in the docs. Will make a new PR soon.

**Comment by bstroebl** [comment]
PR created: https://github.com/qgis/QGIS/pull/3443

**Pull request merged**